platforms/XboxOne/builder.js

```diff
@@ -0,0 +1,77 @@
+const utils = require('../../tasks/utils');
+const path = require("path");
+
+module.exports = {
+    build: build
+};
+
+const pkg = utils.readJSON(path.resolve(__dirname, "../../package.json"));
+
+function build(env, serverUrl) {
+    if (!env) env = 'dev';
+    if (!serverUrl) serverUrl = utils.appUrl;
+
+    const version = utils.ensureVersion(pkg.version, 4);
+
+    const appName = `${pkg.name}_xboxone_${env}_${version}`;
+    console.log(`building ${appName}`);
+
+    if (process.platform != "win32") {
+        utils.fatalError("XboxOne builds can only be done on a Windows 10 machine.");
+    }
+
+    const windowsAppName = 'RefApp';
+    const configDir = path.resolve('platforms/XboxOne', windowsAppName);
+    const configPath = path.resolve(configDir, windowsAppName + ".csproj");
+
+    
+    // Update the version for XboxOne
+    const packageManifest = path.resolve(configDir, "Package.appxmanifest");
+    utils.replacePatterns(packageManifest, [
+        {
+            match: /Version=".*/,
+            replacement: 'Version="' + version + '" />'
+        },
+    ]);
+
+    const propertiesDir = path.resolve(configDir, "Properties");
+
+    const assemblyInfo = path.resolve(propertiesDir, "AssemblyInfo.cs");
+    utils.replacePatterns(assemblyInfo, [
+        {
+            match: /\[assembly: AssemblyVersion.*/,
+            replacement: '[assembly: AssemblyVersion("' + version + '")]'
+        },
+        {
+            match: /\[assembly: AssemblyFileVersion.*/,
+            replacement: '[assembly: AssemblyFileVersion("' + version + '")]'
+        },
+    ]);
+
+    const msBuildArgs = [configPath,
+        "/p:Platform=x64",
+        "/p:PlatformTarget=x64",
+        "/p:AppxBundlePlatforms=x64",
+        "/p:Configuration=Release",
+        "/p:AppxBundle=Always",
+        "/p:DebugType=pdbonly",
+        "/p:DebugSymbols=false"];
+
+    const distDir = path.resolve(__dirname, '../../dist');
+    utils.mkDir(distDir);
+
+    // Path to MSBuild may change developer to developer
+    const msBuildExe = path.resolve('D:/Program Files (x86)/Microsoft Visual Studio/2019/Professional/MSBuild/Current/Bin', 'MSBuild.exe');
+    
+    utils.spawn(msBuildExe, msBuildArgs);
+    
+    const appPackagesDir = path.resolve(configDir, "AppPackages");
+    const appZipPath = path.resolve(distDir, appName + ".zip");
+    const bundleDirPath = path.resolve(appPackagesDir, `${windowsAppName}_${version}_Test`);
+    const bundlePath = path.resolve(bundleDirPath, `${windowsAppName}_${version}_x64.msixbundle`);
+
+    utils.zipFile(bundlePath, appZipPath);
```

Comment:
the bundle does not itself need to be compressed i think, it is more convenient to just have the final installer be a direct file available for sideloading

you will want to make a bundleDistPath instead of appZipPath